src/Component/Metadata/Operation.php

```diff
@@ -0,0 +1,474 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Metadata;
+
+/**
+ * @experimental
+ */
+class Operation
```

Comment:
My generic opinion is that it would be easier to iterate over this feature if it will be split into even smaller groups. Not all 34 properties injected into this class will be used at once and this seems like a straight road into dead/unused attributes in the future. Can we introduce Operation with the first usage? It doesn't need to be feature complete. 

But also, let me review rest of the PRs